internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -251,41 +261,117 @@ func convertEFSPermissions(input map[string]manifest.Volume) ([]*template.EFSPer
 		if volume.EFS == nil {
 			continue
 		}
+		if volume.EFS.UseManagedFS() {
+			continue
+		}
 		// Write defaults to false.
 		write := defaultWritePermission
 		if volume.ReadOnly != nil {
 			write = !aws.BoolValue(volume.ReadOnly)
 		}
-		if volume.EFS.FileSystemID == nil {
+
+		id, err := volume.EFS.FSID()
+		if err != nil {
+			return nil, err
+		}
+		if id == nil {
 			return nil, errNoFSID
 		}
+
 		var accessPointID *string
-		if volume.EFS.AuthConfig != nil {
-			accessPointID = volume.EFS.AuthConfig.AccessPointID
+		if volume.EFS.Config.AuthConfig != nil {
+			accessPointID = volume.EFS.Config.AuthConfig.AccessPointID
 		}
 		perm := template.EFSPermission{
 			Write:         write,
 			AccessPointID: accessPointID,
-			FilesystemID:  volume.EFS.FileSystemID,
+			FilesystemID:  id,
 		}
 		output = append(output, &perm)
 	}
 	return output, nil
 }
 
-func convertVolumes(input map[string]manifest.Volume) ([]*template.Volume, error) {
-	if len(input) == 0 {
-		return nil, nil
+func convertManagedFSInfo(wl manifest.Workload, input map[string]manifest.Volume) (*template.ManagedVolumeCreationInfo, error) {
+	var output *template.ManagedVolumeCreationInfo
+	for name, volume := range input {
+		if output != nil {
```

Comment:
Should we still return an error when `output != nil` if  `volume.EFS == nil` or `volume.EFS.UseManagedFS() == false` ?

Reply:
This should only be triggered now if there is more than one volume using Managed EFS in the list. Good catch.